test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -44,10 +44,35 @@ class test_uct_sockaddr : public uct_test {
                           server_recv_req(0), delay_conn_reply(false) {
     }
 
-    void init() {
-        if (!strcmp(GetParam()->md_name.c_str(), "tcp")) {
-            UCS_TEST_SKIP_R("tcp does not support sockaddr over iface");
+    void check_md_usability() {
+        uct_md_attr_t md_attr;
+        uct_md_config_t *md_config;
+        ucs_status_t status;
+        uct_md_h md;
+
+        status = uct_md_config_read(GetParam()->component, NULL, NULL, &md_config);
+        EXPECT_TRUE(status == UCS_OK);
+
+        status = uct_md_open(GetParam()->component, GetParam()->md_name.c_str(),
+                             md_config, &md);
+        EXPECT_TRUE(status == UCS_OK);
+        uct_config_release(md_config);
+
+        status = uct_md_query(md, &md_attr);
+        ASSERT_UCS_OK(status);
+
+        uct_md_close(md);
+
+        if (!(md_attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) {
+            UCS_TEST_SKIP_R(GetParam()->md_name.c_str() +
```

Comment:
we need to instantiate the test only on MDs with sockaddr, in the first place. instead of skipping on non-supporting MDs

Reply:
you mean to have different resources for the two sockaddr classes in this file?
is this really required especially if we plan to remove this code?

Reply:
i mean, we now instantiate the tests for mds with UCT_MD_FLAG_SOCKADDR and for components with UCT_COMPONENT_FLAG_CM so both go into each of the two class in this file.
tcp only has the CM flag which is needed for the cm_testing. rdmacm has both.

Reply:
if we can remove this when old sockaddr code is removed, then it's ok